### PR TITLE
Amd aocc support

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
Added AMD AOCC Compiler support to WRF 

TYPE: choose one of [new feature]

KEYWORDS: AMD, AOCC, Compiler Support, ZEN Architecture, Configure

SOURCE:  amd-toolchain-support@github.com 

DESCRIPTION OF CHANGES:
Problem: Added AMD AOCC Compiler Support
Earlier there is no support for AMD AOCC Complier. Now we have added support. 

Solution:
Added AMD AOCC Compiler support to configure using stanzas. 

LIST OF MODIFIED FILES: 
M       arch/configure.defaults

TESTS CONDUCTED: 
1. Compiled and tested with AOCC Compiler. Functional test with AOCC 3.2.0 is done.
2. Passed regression tests.

RELEASE NOTE: Provided AMD AOCC Compiler support at the time of configuring WRF for AMD Architectures including Zen1, Zen2 and Zen3.

### CODE REVIEW

**Comment by weiwangncar** [comment]
Jenkins tests have passed:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           58
    Number of Simulations  : 158           156        0
    Number of Comparisons  : 95           92        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Comment by amd-toolchain-support** [comment]
https://github.com/spack/spack/pull/29106

**Comment by alalazo** [comment]
@amd-toolchain-support Why was this PR closed? Is there no interest in having WRF compile with `aocc`?

**Comment by amd-toolchain-support** [comment]
Hi @weiwangncar, 
Thanks for your review comments. We have added real*8 support and compiled both WRF Chem and WRF DA successfully along with WRF with `AMD AOCC Compiler 3.2` on `Cent OS 8`. 

Request you to do the needful and share comments.

**Comment by weiwangncar** [comment]
Jenkins tests have passed:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           58
    Number of Simulations  : 158           156        0
    Number of Comparisons  : 95           92        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

File: arch/configure.defaults
Comment (amd-toolchain-support) [reply arch/configure.defaults:45]: Hi @weiwangncar,

Thanks for your review comments. We have address this NETCDFPAR_BUILD flag issues and updated in our latest commit [6a72ef9]

Request you to do the needful

Comment (amd-toolchain-support) [reply arch/configure.defaults:45]: Hi @weiwangncar,

A gentle reminder. 

We have addressed this NETCDFPAR_BUILD flag issues and updated in our latest commit [https://github.com/wrf-model/WRF/commit/6a72ef913976e50c21f3f244c09e1b6d28501aff]. Requesting you to review.

Thank you

Comment (weiwangncar) [reply arch/configure.defaults:45]: @amd-toolchain-support Thanks for doing your part. We will be reviewing and approving it soon.

Comment (amd-toolchain-support) [reply arch/configure.defaults:45]: Hi @weiwangncar ,

I hope you have some positive update on our submission. 

Thank you.

File: arch/configure.defaults
Comment (mgduda) [inline arch/configure.defaults:2070]: Usually this line includes some information about the compilers that are used in the configuration. Should something like `LLVM` or `Flang/Clang` or something similar be included?

Comment (amd-toolchain-support) [reply arch/configure.defaults]: Hi @mgduda ,

Thanks for your comments. We have addressed your comments in the [d39e2ca](https://github.com/wrf-model/WRF/pull/1684/commits/d39e2cadaa7618a0ee8f7f03ce9dcf9a62776fce) .

Request you to do the needful.

Thank you

Comment (amd-toolchain-support) [reply arch/configure.defaults]: Hi @mgduda and @weiwangncar,

Gentle Reminder.

It has been a long time since we have not received any update. Could you please look at [d39e2ca](https://github.com/wrf-model/WRF/pull/1684/commits/d39e2cadaa7618a0ee8f7f03ce9dcf9a62776fce) and confirm?

Thanks in advance.

Comment (mgduda) [reply arch/configure.defaults]: @amd-toolchain-support I just left a comment in the other thread on this PR regarding whitespace on line 45.

**Comment by weiwangncar** [comment]
@mgduda What is still needed for this PR?

**Review by mgduda — CHANGES_REQUESTED** [review]
Can you please revert the change to line 45 of `arch/configure.defaults`?

**Comment by weiwangncar** [comment]
@amd-toolchain-support Can you respond to the comment by mgduda?

**Comment by amd-toolchain-support** [comment]
@mgduda thank you for your response. We are currently finalizing the support for AOCC 4.0 and Zen4 ARCH in WRF internally. Please give us some time to update this PR.

**Comment by mgduda** [comment]
> @mgduda thank you for your response. We are currently finalizing the support for AOCC 4.0 and Zen4 ARCH in WRF internally. Please give us some time to update this PR.

Although I'm not speaking officially on behalf of the WRF development group, I believe the plan is to create the `release-v4.5` branch from `develop` in the next day or two. So, it might be best if you could revert the change to line 45 in order for us to merge these changes before the creation of the release branch. It sounds like any changes for AOCC 4.0 and Zen4 may need to wait until the next WRF release.

**Comment by amd-toolchain-support** [comment]
@mgduda We have reverted the change introduced in Line 45. Please merge this PR which adds the support for AOCC 3 in `release-v4.5`. We will create a new Pull Request for AOCC 4 support.

**Comment by weiwangncar** [comment]
The Jenkins test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Comment by weiwangncar** [comment]
@amd-toolchain-support Thanks for making the change.
@mgduda Can you review it and approve it if it is good?

**Review by mgduda — APPROVED** [review]

**Review by weiwangncar — APPROVED** [review]
Does not affect other parts of the code.

**Comment by weiwangncar** [comment]
@amd-toolchain-support Not sure if you have done so, but the option should be tested for WRF-Chem, and WRFDA. If you have done so, please add the info to the PR.

**Comment by amd-toolchain-support** [comment]
Hi @weiwangncar, 

We have successfully compiled both WRFDA and WRF-Chem and generated required executables without any errors.

**Comment by weiwangncar** [comment]
@amd-toolchain-support Thank you for letting us know.

**Comment by ravitejakuppili** [comment]
Hi,

The recipe which is available works with AOCC 3.
https://github.com/wrf-model/WRF/pull/1684#issuecomment-1406135759 

We are working on new recipe to support AOCC 4.

**Comment by weiwangncar** [comment]
@ravitejakuppili Thanks for your comments and clarification.

**Pull request merged**
